### ISSUE TITLE
[OpenCV.js] Error when creating VideoCapture : wrong video size.

### ISSUE THREAD

**Issue by theloniouss** [opencv/opencv#19922]
##### System information (version)

- OpenCV => 4.5.2
- Operating System / Platform => Chrome on Android
- Compiler => :grey_question:

##### Detailed description

Hi ! I'm trying to use OpenCV.js on mobile phones for image tracking. For now I follow the tutorials to get the basics. I've set a \<video\> tag that starts streaming a camera on the mobile phone, then I create a new VideoCapture to capture the stream. The problem is that VideoCapture seams to check the _video.height_ and _video.width_ properties that are not initialized after the stream starts. Is it an error in the tutorial or did I miss something ?

Link to the tutorial : [https://docs.opencv.org/4.5.2/dd/d00/tutorial_js_video_display.html](https://docs.opencv.org/4.5.2/dd/d00/tutorial_js_video_display.html)

##### Steps to reproduce

What I did is to wait for the first frame to be rendered with _setTimeout_ because its size is (0, 0) and then I change _video.height_ and _video.width_ by hand before creating VideoCapture :
```js
// start video stream

setTimeout(function () {
    video.height = video.videoHeight;
    video.width = video.videoWidth;
    cap = new cv.VideoCapture(video);

    // call to process function here
}, 50);
```

##### Issue submission checklist

 - [X] I report the issue, it's not a question
   
 - [X] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [X] I updated to latest OpenCV version and the issue is still there
   
 - [X] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by Chaphasilor** [comment]
yeah, the implementation seems to be wrong. it's checking the "resolution" of the provided video element again for some reason (why do we have to pass it in the first place then?), but they use `<video element>.width` (with is the size of the HTML element, *not* the resolution) instead of `<video element>.videoWidth` (and the same for `height`).  
I came up with the same workaround :)

**Comment by neissljanik** [comment]
Another workaround is to `await` the video's `onloadedmetadata` event, the video's dimensions are accessible after said event fires (https://stackoverflow.com/a/9333276). So, the complete workaround could look like this:

```js
await new Promise(resolve => video.onloadedmetadata = () => resolve());

const width = video.videoWidth, height = video.videoHeight;
video.width = width;
video.height = height;

const cap = new cv.VideoCapture(video);
```
